heron/common/src/python/handler/access/heron.py

```diff
@@ -122,36 +122,64 @@ def get_cluster_topologies(cluster):
   raise tornado.gen.Return((yield fetch_url_as_json(request_url)))
 
 ################################################################################
+# Get the list of topologies given a cluster submitted by a given role
+################################################################################
+@tornado.gen.coroutine
+def get_cluster_role_topologies(cluster, role):
+  endpoint = create_url(TOPOLOGIES_URL_FMT)
+  request_url = tornado.httputil.url_concat(endpoint,
+    dict(cluster=cluster, role=role))
+  raise tornado.gen.Return((yield fetch_url_as_json(request_url)))
+
+################################################################################
+# Get the list of topologies given a cluster submitted by a given role under
+# a given environment
+################################################################################
+@tornado.gen.coroutine
+def get_cluster_role_env_topologies(cluster, role, env):
+  endpoint = create_url(TOPOLOGIES_URL_FMT)
+  request_url = tornado.httputil.url_concat(endpoint,
+    dict(cluster=cluster, role=role, environ=env))
+  raise tornado.gen.Return((yield fetch_url_as_json(request_url)))
+
+################################################################################
 # Get the execution state of a topology in a cluster
 ################################################################################
 @tornado.gen.coroutine
-def get_execution_state(cluster, environ, topology):
+def get_execution_state(cluster, environ, topology, role=None):
   request_url = tornado.httputil.url_concat(
       create_url(EXECUTION_STATE_URL_FMT),
       dict(cluster = cluster, environ = environ, topology = topology)
   )
+  if role:
+    request_url = tornado.httputil.url_concat(request_url, dict(role=role))
+
```

Comment:
This was the first way of refactoring I tried. I believe this breaks consistency because these pieces of code use ``url_concat`` on optional parameters: https://github.com/twitter/heron/blob/master/heron/ui/src/python/handlers/access/heron.py#L324